.github/workflows/build-golang-fedora.yaml

```diff
@@ -0,0 +1,117 @@
+# (C) Copyright Confidential Containers Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+name: "build golang-fedora container image"
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "hack/Dockerfile.golang"
+      - "hack/update-go-container.sh"
+  workflow_dispatch:
+    inputs:
+      registry:
+        description: "The container registry to push the image to. Must be either 'ghcr.io' or 'quay.io'"
+        default: "quay.io"
+      image_repository:
+        description: "The name of the container image repository to push the image to."
+        default: "confidential-containers/cloud-api-adaptor"
```

Comment:
oh, one more thing @katexochen , I promise :)

I created a public `quay.io/wainersm/cloud-api-adaptor`  where I published the built CAA image as `quay.io/wainersm/cloud-api-adaptor/cloud-api-adaptor:dev-cca53f961f11ecace745b6132aef51910abbb68e-dirty`.  When I tried to install CAA on my cluster I got authorization denied to pull the image. I could only get it properly working when I re-published the image in one level down of the repository hierarch (i.e. `quay.io/wainersm/cloud-api-adaptor:dev-cca53f961f11ecace745b6132aef51910abbb68e-dirty`).

That said, I am worried that pushing the builder image to `quay.io/confidential-containers/cloud-api-adaptor/golang-fedora:<golang>-<fedora>` we will hit the same issue.

Reply:
Updated it to `quay.io/confidential-containers/golang-fedora:<golang>-<fedora>`

Reply:
> Is this a known limitation of quay.io? I guess another good reason for switching to ghcr...

I don't know. I had never tried to publish to `quay.io/<org>/<repo>/<image>` before.

Reply:
Yeah - I hit this problem with creating the early coco test images, which I failed to explain properly earlier in this PR